HandsOnProject/app/src/main/java/com/example/handsonproject/MainActivity.kt

```diff
@@ -1,11 +1,144 @@
 package com.example.handsonproject
 
-import androidx.appcompat.app.AppCompatActivity
+
 import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.example.handsonproject.databinding.ActivityMainBinding
+import com.example.handsonproject.databinding.ActivityMainBinding.inflate
+
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    private var op : Int = 0
+
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        binding = inflate(LayoutInflater.from(this))
+        setContentView(binding.root)
+        with(binding){
+            editText1.hint="angka"
+            textView.text=""
+        }
+    }
+
+
+    fun onTapB1(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "1") }
+    }
+    fun onTapB2(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "2") }
+    }
+    fun onTapB3(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "3") }
+    }
+    fun onTapB4(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "4") }
+    }
+    fun onTapB5(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "5") }
+    }
+    fun onTapB6(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "6") }
+    }
+    fun onTapB7(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "7") }
+    }
+    fun onTapB8(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "8") }
+    }
+    fun onTapB9(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "9") }
+    }
+    fun onTapB0(view: View){
+        with(binding) { editText1.setText(editText1.getText().toString() + "0") }
+    }
+    fun onTapBEqual(view: View){
+        with(binding){
+            if (editText1.text.toString().equals("")&&textView.text.toString().equals("")) return
+            if(op==1){
+                val res=editText1.text.toString().toDouble()+textView.text.toString().toDouble()
+                editText1.setText(res.toString())
+                textView.setText("")
```

Comment:
@shovian It would be good to use variable names that make it clear what the difference is between "editText1" and "textView". I assume one of them would be the "input" and the other one the "result", so maybe you could rename them to reflect that.

Reply:
Was implemented